File: Matlab/matvec/matvec_row.m
function b = matvec_row(A, x) 
% Usage: b = matvec_row(A, x) 
%
% Function to perform row-oriented matrix multiplication
% 
% Inputs: 
%    A is a matrix (m x n)
%    x is a vector (n)
%
% Outputs:
%    b is a vector (m)
%
% Daniel R. Reynolds
% SMU Mathematics
% Math 5316
% Spring 2019

% get problem dimensions
[m,n] = size(A);

% check that A and x are compatible
if (n ~= length(x))
   error('matvec_row error: A and x are incompatible')
end

% initialize output
b = zeros(m,1);

% perform product
for i=1:m
   for j=1:n
      b(i) = b(i) + A(i,j)*x(j);
   end
end

File: Python/matvec/matvec_row.py
# matvec_row.py
#
# Daniel R. Reynolds
# SMU Mathematics
# Math 5316
# Spring 2019

def matvec_row(A, x):
    """
    Usage: b = matvec_row(A, x) 

    Function to perform row-oriented matrix multiplication
 
    Inputs: 
       A is a matrix (m x n numpy matrix)
       x is a vector (n numpy array)

    Outputs:
       b is a vector (m numpy array)
    """

    # imports
    import numpy as np

    # get problem dimensions
    m, n = np.shape(A)
    
    # check that A and x are compatible
    if (n != x.size):
        raise ValueError("matvec_row error: A and x are incompatible")
        
    # initialize output
    b = np.zeros(m, dtype=float)
    
    # perform product
    for i in range(m):
        b[i] += sum(A[i,:]*x)

    return b


      
